dbms/src/TiDB/Schema/TiDBSchemaSyncer.h

```diff
@@ -180,6 +178,16 @@ struct TiDBSchemaSyncer : public SchemaSyncer
         }
         LOG_FMT_DEBUG(log, "End load schema diffs with total {} entries.", diffs.size());
 
+
+        // Since the latest schema diff may be empty, and schemaBuilder may need to update the latest version for storageDeltaMerge,
+        // Thus we need check whehter latest schema diff is empty or not before begin to builder.applyDiff.
+        if (!diffs.back())
```

Comment:
Maybe we can pop back the last diff here. And also update `used_version` and add some comment here.
And later logic can be changed to something like
```
for (size_t diff_index = 0; diff_index < diffs.size(); ++diff_index)
{
     const auto & schema_diff = diffs[diff_index];
     if (!schema_diff)
     {
          /// some comment
          LOG_FMT_WARNING(log, "Skip the schema diff from version {}. ", cur_version + diff_index + 1);
          continue;
     }
     builder.applyDiff(*schema_diff);
}
return used_version;
```
This just looks more clear to me.

Reply:
Yes, I think it's better, and I change the logic follow this.